upstairs/src/client.rs

```diff
@@ -1858,7 +1849,6 @@ impl DownstairsClient {
                             String::new()
                         },
                     );
-                    self.checked_state_transition(up_state, DsState::New);
```

Comment:
This is a drive-by change: the calls to `restart_connection` below always update the state, so doing it here is superfluous.